.github/workflows/makecode.yml

```diff
@@ -0,0 +1,94 @@
+name: Build MakeCode
+
+# MakeCode has to git clone/checkout this repo commit SHA, so only run this
+# workflow on pushes, as PRs generate a "merge commit" that is not clonable.
+on:
+  push:
+    branches: '*'
+
+jobs:
+  build-makecode:
+    strategy:
+      matrix:
+        # One job builds with the local toolchain, the other with Docker
+        pxt-flags: ["PXT_NODOCKER=1", ""]
+      fail-fast: false
+    name: Build MakeCode ${{ matrix.pxt-flags && '(nodocker)' }}
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Setup Python 3.7
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.7'
+      - name: Setup arm-none-eabi-gcc v10.3
+        if: ${{ matrix.pxt-flags }}
+        uses: carlosperate/arm-none-eabi-gcc-action@v1
+        with:
+          release: 10.3-2021.10
+      - name: Install Yotta, Ninja v1.10 & CMake v3.22 via PyPI
+        if: ${{ matrix.pxt-flags }}
+        # MarkupSafe < 2.0 needed for Yotta
+        run: python -m pip install MarkupSafe==1.1.1 ninja==1.10.2.2 cmake==3.22.1 yotta
```

Comment:
Is MarkupSafe required? I don't see it being used in the actual action?

Reply:
It's required to be lower than version 2.0 for Yotta, as it's brought as a dependencies without an upper version limit.

Reply:
Ah, ok, so I guess the rest just auto-install with yotta. Gotcha, that's fine then.